cmd/argocd/commands/cluster.go

```diff
@@ -106,6 +107,11 @@ func NewClusterAddCommand(clientOpts *argocdclient.ClientOptions, pathOpts *clie
 			contextName := args[0]
 			conf, err := getRestConfig(pathOpts, contextName)
 			errors.CheckError(err)
+			if clusterOpts.ProxyUrl != "" && conf.Proxy == nil {
+				u, err := argoappv1.ParseProxyUrl(clusterOpts.ProxyUrl)
+				errors.CheckError(err)
+				conf.Proxy = http.ProxyURL(u)
+			}
```

Comment:
For the case when the user has both a proxy entry in his kubeconfig, and passes the proxy explicitly in the command line, we probably want to use the command line arg, will this be the behavior in this case?

Reply:
Fixed :)